tests/run-make/suspicious-library/rmake.rs

```diff
@@ -0,0 +1,14 @@
+// This test creates some fake dynamic libraries with nothing inside,
+// and checks if rustc avoids them and successfully compiles as a result.
+
+//@ ignore-cross-compile
+
+use run_make_support::{dynamic_lib, rustc};
+use std::fs::File;
+
+fn main() {
+    rustc().input("foo.rs").arg("-Cprefer-dynamic").run();
+    File::create(dynamic_lib("foo-something-special")).unwrap();
+    File::create(dynamic_lib("foo-something-special2")).unwrap();
+    rustc().input("bar.rs");
```

Comment:
Problem: this needs `.run()` to assert that the command status is success.

Reply:
We should probably apply https://doc.rust-lang.org/beta/reference/attributes/diagnostics.html#the-must_use-attribute to many things in `run-make-support`.

Reply:
Yes, we definitely should add `#[must_use]` to a lot of things.

Reply:
I opened https://github.com/rust-lang/rust/issues/125703 to track this so I don't forget.

Reply:
I'll experiment, but we might even want to add "drop bombs" to ensure `Command` wrappers are actually "used".